src/client/Client.js

```diff
@@ -432,6 +432,9 @@ class Client extends BaseClient {
     if (typeof options.messageSweepInterval !== 'number' || isNaN(options.messageSweepInterval)) {
       throw new TypeError('CLIENT_INVALID_OPTION', 'messageSweepInterval', 'a number');
     }
+    if (typeof options.messageEditHistoryMaxSize !== 'number' || isNaN(options.messageEditHistoryMaxSize)) {
+      throw new TypeError('CLIENT_INVALID_OPTION', 'messageEditHistoryMaxSize', 'a number');
+    }
```

Comment:
```suggestion
    if (typeof options.messageEditHistoryMaxSize !== 'number' || isNaN(options.messageEditHistoryMaxSize) || options.messageEditHistoryMaxSize < -1) {
      throw new TypeError('CLIENT_INVALID_OPTION', 'messageEditHistoryMaxSize', 'a number');
    }
```
Number lower than -1 could mess with `Message#patch`:
```js
    if (this.client.options.messageEditHistoryMaxSize !== -1) {
      this._edits.splice(this.client.options.messageEditHistoryMaxSize);
    }
```

Reply:
Ah yeah, good call